libs/ha/ha_common_libs_k8s.py

```diff
@@ -1441,48 +1443,77 @@ def get_node_resource_ids(node_obj, r_type, n_type=None, node_id=None):
         LOGGER.info("Resource IDs for %s are: %s", r_type, resp)
         return resp
 
-    def delete_single_pod_setting_replica_0(self, master_node_obj, health_obj,
-                                            pod_prefix=common_const.POD_NAME_PREFIX):
+    def delete_kpod_with_shutdown_methods(self, master_node_obj, health_obj,
+                                          pod_prefix=None, kvalue=1,
+                                          down_method=common_const.RESTORE_SCALE_REPLICAS):
         """
-        Delete Single pod by setting replica set to 0. Check service status is in degraded mode.
+        Delete K pods by given shutdown method. Check and verify deleted/remaining pod's services
+        status, cluster status.
         :param master_node_obj: Master node object list
         :param health_obj: Health object
-        :param pod_prefix: Pod prefix to be deleted.
+        :param pod_prefix: Pod prefix to be deleted (Expected List type).
```

Comment:
Why pod prefix is list, any particular reason ?

Reply:
This is because in some scenarios we want to delete both server as well as data pods